File: Pemrograman Kompetitif Dasar/ch03/co11p229.py
#!/bin/python
def read(mode=2):
    inp = input().strip()
    if mode == 0: return inp
    if mode == 1: return inp.split()
    if mode == 2: return list(map(int,inp.split()))
N, temp = read()
A = [read() for i in range(N)]
M, temp = read()
B = [read() for i in range(N)]
if A == B:
    print('identik')
elif A == B[::-1]:
    print('horizontal')
elif A == [i[::-1] for i in B]:
    print('vertikal')
elif A == list(map(list,list(zip(*B)))):
    print('diagonal kanan bawah')
elif A == [i[::-1] for i in map(list,list(zip(*B))[::-1])]:
    print('diagonal kiri bawah')
else:
    print('tidak identik')

File: PKD/ch03/co11p229.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  ios_base::sync_with_stdio(0);
  bitset<6> flag;
  flag.set();
  int N, M, temp;
  cin >> N >> temp;
  int A[N][N];
  for (int i = 0; i < N; i++)
    for (int j = 0; j < N; j++)
      cin >> A[i][j];
  cin >> M >> temp;
  if (N != M) {
    cout << "tidak identik";
  } else {
    for (int i = 0; i < M; i++)
      for (int j = 0; j < M; j++) {
        cin >> temp;
        if (A[i][j] != temp)
          if (flag.test((size_t)0))
            flag.set((size_t)0,0);
        if (A[i][M-j-1] != temp)
          if (flag.test((size_t)1))
            flag.set((size_t)1,0);
        if (A[M-i-1][j] != temp)
          if (flag.test((size_t)2))
            flag.set((size_t)2,0);
        if (A[j][i] != temp)
          if (flag.test((size_t)3))
            flag.set((size_t)3,0);
        if (A[M-j-1][M-i-1] != temp)
          if (flag.test((size_t)4))
            flag.set((size_t)4,0);
      }
    string res[] = {"identik", "vertikal", "horisontal",
                    "diagonal kanan bawah",
                    "diagonal kiri bawah",
                    "tidak identik"};
    for (size_t i = 0; i < 6; i++)
      if (flag.test(i)) {
        cout << res[i] << "\n";
        break;
      }
  }
  return 0;
}
